src/contrib/cluster/Akka.DistributedData/DeltaPropagationSelector.cs

```diff
@@ -55,89 +57,83 @@ public void Delete(string key)
                         
-                        var j = deltaSentToNodeForKey.GetValueOrDefault(node, 0L);
-                        var deltaEntriesAfterJ = DeltaEntriesAfter(entries, j);
-                        if (!deltaEntriesAfterJ.IsEmpty)
-                        {
-                            var fromSeqNr = deltaEntriesAfterJ.Keys.First(); // should be min
-                            var toSeqNr = deltaEntriesAfterJ.Keys.Last(); // should be max
+                    var j = deltaSentToNodeForKey.GetValueOrDefault(node, 0L);
+                    var deltaEntriesAfterJ = DeltaEntriesAfter(entries, j);
+                    if (!deltaEntriesAfterJ.IsEmpty)
+                    {
+                        var fromSeqNr = deltaEntriesAfterJ.Keys.First(); // should be min
+                        var toSeqNr = deltaEntriesAfterJ.Keys.Last(); // should be max
 
-                            // in most cases the delta group merging will be the same for each node,
-                            // so we cache the merged results
-                            var cacheKey = (key, fromSeqNr, toSeqNr);
-                            if (!cache.TryGetValue(cacheKey, out var deltaGroup))
+                        // in most cases the delta group merging will be the same for each node,
+                        // so we cache the merged results
+                        var cacheKey = (key, fromSeqNr, toSeqNr);
+                        if (!cache.TryGetValue(cacheKey, out var deltaGroup))
+                        {
+                            deltaGroup = deltaEntriesAfterJ.Values.Aggregate((d1, d2) =>
                             {
-                                using (var e = deltaEntriesAfterJ.Values.GetEnumerator())
-                                {
-                                    e.MoveNext();
-                                    deltaGroup = e.Current;
-                                    while (e.MoveNext())
-                                    {
-                                        deltaGroup = deltaGroup.Merge(e.Current);
-                                        if (deltaGroup is IReplicatedDeltaSize s && s.DeltaSize > MaxDeltaSize)
-                                        {
-                                            deltaGroup = DeltaPropagation.NoDeltaPlaceholder;
-                                        }
-                                    }
-                                }
-
-                                cache[cacheKey] = deltaGroup;
-                            }
-
-                            deltas[key] = (deltaGroup, fromSeqNr, toSeqNr);
-                            _deltaSentToNode = _deltaSentToNode.SetItem(key, deltaSentToNodeForKey.SetItem(node, toSeqNr));
+                                var merged = ReferenceEquals(d2, DeltaPropagation.NoDeltaPlaceholder) 
+                                    ? DeltaPropagation.NoDeltaPlaceholder 
+                                    : d1.Merge(d2);
```

Comment:
This is the same line on the new code that the bug might happen again, but now it is guarded with a reference check